resolver/resolver_test.go

```diff
@@ -242,7 +252,26 @@ func TestHandler(t *testing.T) {
 	if m.Rcode != 3 {
 		t.Error("not setting NXDOMAIN for AAAA requests")
 	}
+}
+
+func TestHandlerWithDomainPatterns(t *testing.T) {
+	res, err := fakeDNSwithPatterns(8053, &[]string{"test-{{.tname}}"})
+	if err != nil {
+		t.Error(err)
+	}
 
+	dns.HandleFunc("mesos.", res.HandleMesos)
+	go res.Serve("udp")
+	go res.Serve("tcp")
+
+	// wait for startup ? lame
```

Comment:
there's a new way to wait for DNS server startup (`safeStartDNSResolver`) coming down the line in #144.
should probably do the same here once that PR has been merged.